### DISCUSSION TITLE
Why not Redux Toolkit?

### DISCUSSION THREAD

**Discussion by AfrazHussain** [infinitered/ignite#1949] [General]
Firstly, to whoever contributed to this boilerplate: THANK YOU! 

TLDR: I get the argument of using MST over Redux (as stated in the docs), but what about [Redux Toolkit](https://redux-toolkit.js.org/) + [RTK Query](https://redux-toolkit.js.org/rtk-query/overview) over MST?

A couple of years ago, I started a RN project using the Ignite Andross boilerplate. It had Redux, which as the project scaled, became impossible to properly manage, along with all of its other drawbacks and missing features. At the same time, Ignite released the Bowser boilerplate, which had MST, but we didn't switch over just because the project built with Redux was too large.

Fast forward to today, I've worked on couple of projects with Redux Toolkit and I absolutely love it. My question is: are there any particular benefits of having MST instead of Redux Toolkit? Performance, features, etc?

**Comment by ryskin** [comment]
https://github.com/infinitered/ignite/blob/master/docs/MobX-State-Tree.md

**Comment by jamonholmgren** [comment]
The document that @ryskin linked to explains why we like mobx-state-tree (MST), but your question is a good one, @AfrazHussain.

It's important to remember that Ignite is an expression of how we at Infinite Red like to build React Native apps, based on many years of RN experience (we started in October of 2015 and have been building in RN ever since).

So far, our experience has been generally better when using MST over Redux, including some projects using Redux Toolkit. We build in a variety of state management solutions for clients, based on their needs, but when we have choice, we almost always choose to use MST still to this day.

So the answer is simply "Because IR likes it better". :-)

I know other coders legitimately like RTK better, and that's all good. No problem at all. But as long as IR continues to prefer MST, we'll continue to maintain our boilerplate with that state management decision.

With that said, we have some plans to revisit our state management solution sometime this summer. So, stay tuned.

![because of reasons](https://c.tenor.com/d-5TwKQ08c4AAAAd/because-becauseofreasons.gif)

**Reply by markerikson** [reply]
I'm the creator and co-maintainer of RTK, and I endorse this answer :)

**Reply by phryneas** [reply]
And another co-maintainer, endorsing it likewise ^^

But I recently I looked into Ignite and stumbled over the [why not Redux](https://github.com/infinitered/ignite/blob/master/docs/MobX-State-Tree.md) page - it would be great if you could update that a bit, because it's a lot less nuanced and just gave me the first impression "they're clearly not up to date there".

**Reply by IvanFestina** [reply]
Hi, did you revisti your desition last summer ?)
